go_modules/helpers/updater/helpers.go

```diff
@@ -6,7 +6,11 @@ import (
 	"golang.org/x/mod/modfile"
 )
 
-// Private methods lifted from the `modfile` package
+// Private methods lifted from the `modfile` package.
+// Last synced: 4/28/2021 from:
+// https://github.com/golang/mod/blob/858fdbee9c245c8109c359106e89c6b8d321f19c/modfile/rule.go
+
+var slashSlash = []byte("//")
 
 // setIndirect sets line to have (or not have) a "// indirect" comment.
 func setIndirect(line *modfile.Line, indirect bool) {
```

Comment:
Question (maybe for @jurre): do we still need to do this?

Since https://github.com/dependabot/dependabot-core/pull/2551 , I'd expect `go mod tidy` to detect and handle `// indirect` in the modfile - and probably do a better job than this...

Is this potentially dead code?

Reply:
Yeah I think it's mostly safe to drop this, there are (in theory at least?) some edge-cases in which `go mod tidy` will silently fail and we ignore that, I'm not quite sure what those scenario's are though. In that case this code would still add the `// indirect` comment, but it's unclear to me how valuable that is if `go mod tidy` has already failed

Reply:
I took at timeboxed look at this: [this test](https://github.com/dependabot/dependabot-core/blob/552be5d0236078c18574bba4baa59aa9284da56e/go_modules/spec/dependabot/go_modules/file_updater_spec.rb#L147-L149) started failing.

That harness under test is a little garbage-in-garbage-out. Since https://github.com/dependabot/dependabot-core/pull/3456 , it's invalid:
```
~/git/github/dependabot-core/go_modules/spec/fixtures/projects/repo_not_found $ go run .
go: github.com/go-openapi/spec@v0.19.2 requires
	github.com/go-openapi/jsonpointer@v0.19.2: missing go.sum entry; to add it:
	go mod download github.com/go-openapi/jsonpointer
```

Given the expected error caused by https://github.com/kubernetes/kubernetes/blob/2e808b7cb054ee242b68e62455323aa783991f03/go.mod#L104 , maybe that harness should use `k8s.io/kubernetes` and not worry about `github.com/go-openapi/spec`? 🤷 

I created https://github.com/dependabot-fixtures/go-modules-public to potentially simplify the harness to 3 dependencies, but that fails with a different error (the `PrivateSourceAuthenticationFailure` error is raised in both cases).

I'm just going to merge this update PR and retreat.

Reply:
@thepwagner Do you want to at least file an issue tracking this failure and including a link to your investigation so far? No sense making the next person have to duplicate your initial work here...

Reply:
@jeffwidman I'm not sure it's a change we should pursue: I suspected the code isn't necessary, but it seems to be.

Reply:
Ah right, I misread your comment as investigating a test failure on my _original_ change, didn't realize the test failure was on your experiment. Makes complete sense.